torch/_dynamo/variables/dicts.py

```diff
@@ -623,11 +623,14 @@ def create(cls, user_cls, args, kwargs, options):
             bound = inspect.signature(user_cls).bind(*args, **kwargs)
             bound.apply_defaults()
             raw_items = bound.arguments
+        elif not args:
```

Comment:
Note that this needs to run after the dataclass logic in order to be correct for diffusers.utils.BaseOutput.

Why? Because of dataclass __init__ handling, the dataclass logic dominates keyword-arg ordering when determining dict ordering, so we need to prefer the inspect.signature handler when the custom class happens to be one of the hybrid dataclass / dict types that some libraries use.

Example session demonstrating why the dataclass logic must take precedence:

from diffusers.utils import BaseOutput

@dataclasses.dataclass
class CustomModelOutput(BaseOutput):
    x: str
    y: str
    
cbo = CustomBaseOutput(y=10, x=9)
and at this point cmo.items() will product odict_items([('x', 9), ('y', 10)]).